lib/Sema/CSApply.cpp

```diff
@@ -7527,6 +7527,39 @@ Expr *ConstraintSystem::applySolution(Solution &solution, Expr *expr,
     return nullptr;
   }
 
+  for (auto &e : solution.Conformances) {
+    const auto &conformance = e.second;
+
+    auto anchor = expr;
+    if (auto *locator = e.first)
+      anchor = locator->getAnchor();
+
+    if (conformance.isAbstract())
+      continue;
+
+    auto *concrete = conformance.getConcrete();
+    // If conformance check is not yet complete let's re-check using 'Used'
+    // to require conformance check to happen.
```

Comment:
I could only find one use for `applySolution` it's called by `TypeChecker::typeCheckExpression`